Storybook for dashboard

michalsmiarowski opened the pull request:
Closes: #2439

Adds Storybook for most of the dashboard's components

michalsmiarowski committed:
Install storybook

Add storybook to package.json

michalsmiarowski committed:
Create stories for components

Create stories for some of the components for storybook

michalsmiarowski committed:
Add decorator to TopUpsDataTable story

michalsmiarowski committed:
Create stories for components v2

Create more stories for the components

michalsmiarowski committed:
Remove sidebar from storybook

We are removing the sidebar from css for storybook

michalsmiarowski committed:
Add stories for components v3

Add more stories for the components

michalsmiarowski committed:
Create stories for components v4

michalsmiarowski committed:
Remove unused prop from Undelegations component

I am removing the `title` props from Undelegations page since it is not used.

michalsmiarowski committed:
Add custom center wrapper

Add custom div wrapper for all stories. We are doing that because addon-centered
is little bugged (or storybook) - when saving changes in story and doing hot
reloading the decorator is not added until we refresh page. This custom center
wrapper should fix that.

michalsmiarowski committed:
Create custom decorator that centers the story

The `centered` decorator from the addon is not suitable for some of the
components (for example KeepOnlyPool) so I created the custom one the can be
used per story.

michalsmiarowski committed:
Center all stories properly

Get rid of the storiesOf(...).addDecorator(centered) and instead use decorators:
centered. This will get rid of the bug that the centered addon is not applied
when hot reloading.

michalsmiarowski committed:
Write stories for components with hooks

For components with hooks we are writing a separate component
<component_name>View where we pass data fetched from hooks. This way we can
easily write a story for a view component.

michalsmiarowski committed:
Add storybook-formik addon

michalsmiarowski committed:
Fix stories for components that use Formik

michalsmiarowski committed:
Fix stories with fetched data

michalsmiarowski committed:
Fix storybook warnings

Comment out unused imports

r-czajkowski commented on solidity/dashboard/src/stories/WalletConnectModal.stories.js:
```diff
@@ -0,0 +1,25 @@
+
+export default {
+  title: "WalletConnectModal",
+  component: WalletConnectModal,
```
Each story for modal has transparent background. Can we set white background by default?

r-czajkowski commented on solidity/dashboard/src/stories/Web3Status.stories.js:
```diff
@@ -0,0 +1,35 @@
+export const Connected = Template.bind({})
+Connected.args = {
+  yourAddress: "0xeF42ac774dD0d3519E7CBFD59F36e52038D4e255",
+  isConnected: true,
```
If we set `isConnected` to `true`, the `network disconnected` is displayed. Can we fix it?

r-czajkowski commented on solidity/dashboard/src/stories/TransactionIsPendingMsgContent.stories.js:
```diff
@@ -0,0 +1,19 @@
+
+export default {
+  title: "TransactionIsPendingMsgContent",
+  component: TransactionIsPendingMsgContent,
```
This component is no longer used- so I think we can remove this story and get rid of this component from the dapp.

r-czajkowski commented on solidity/dashboard/src/stories/ApplicationBox.stories.js:
```diff
@@ -0,0 +1,37 @@
+
+export default {
+  title: "ApplicationBox",
+  component: ApplicationBox,
```
The `ApplicationBox` is also no longer used. Let's remove story and component.

r-czajkowski commented on solidity/dashboard/src/stories/TBTCRewardsDataTable.stories.js:
```diff
@@ -0,0 +1,103 @@
+
+export default {
+  title: "TBTCRewardsDataTable",
+  component: TBTCRewardsDataTable,
```
Transparent background- it also applies to some other data tables components eg. `SlashedTokensList`.

r-czajkowski reviewed (changes requested):
Left couple of notes. Overall looking good. I would just consider removing the bigger components or create a kind of hierarchy. Eg `components` for the smallest components in the dapp, `views/templates` for the bigger one eg. forms. I was wondering about the naming convention of the atomic design (atoms/molecules/organisms etc), but that would require a rebuild of almost the entire dapp.

michalsmiarowski committed:
Remove TransactionIsPendingMsgContent component

TransactionIsPendingMsgContent component is not longer used so we are removing
it and it's story from storybook

michalsmiarowski committed:
Remove ApplicationBox component

The ApplicationBox component is no longer used so we are removing it and its
story in storybook.

michalsmiarowski committed:
Add background where needed

Add white or black background to specific stories.

r-czajkowski reviewed (commented):
Shouldn't we add `storybook-static` dir genereted by `npm run build-storybook` to the `gitignore`?

michalsmiarowski committed:
Comment out stories that use injected connector

For now we are commenting out the components that use `injected` connector,
becuase it seems to be a problem when using MetaMask v9.6 and above - there is
an infinite loop which prevents stories from loading.

michalsmiarowski committed:
Add storybook-static dir to .gitignore

Add the /storybook-static directory, which is created when executing
`build-storybook` script, to .gitignore.

r-czajkowski reviewed (approved):
LGTM! Left a final approval for @Battenfield
